packages/core/src/components/tree-view/tree-view.tsx

```diff
@@ -0,0 +1,214 @@
+import { h, Component, ComponentInterface, Event, EventEmitter, Prop } from '@stencil/core';
+
+import { TreeViewItem } from '@dso-toolkit/sources';
+import { DsoTreeItem } from './tree-item';
+
+@Component({
+  tag: 'dso-tree-view',
+  styleUrl: './tree-view.scss',
+  shadow: true
+})
+export class TreeView implements ComponentInterface {
+   /**
+   * The collection of TreeViewItems
+   */
+  @Prop()
+  collection!: TreeViewItem<string>[];
+
+  /**
+   * Emitted when a tree view item is opened.
+   * The `detail` property of the `CustomEvent` will contain the complete path of TreeViewItems from the
+   * root to the item that is emitting the open event. The consumer of the event is responsible for updating
+   * the TreeView's collection (usually set the open state on the last TreeViewItem in path).
+   */
+  @Event()
+  openItem!: EventEmitter<TreeViewItem<string>[]>;
+
+  /**
+   * Emitted when a tree view item is closed.
+   * The `detail` property of the `CustomEvent` will contain the complete path of TreeViewItems from the
+   * root to the item that is emitting the close event. The consumer of the event is responsible for updating
+   * the TreeView's collection (usually set the closed state on the last TreeViewItem in path).
+   */
+  @Event()
+  closeItem!: EventEmitter<TreeViewItem<string>[]>;
+
+  /**
+   * Emitted when a tree view item is clicked.
+   * The `detail` property of the `CustomEvent` will contain the complete path of TreeViewItems from the
+   * root to the item that is emitting the clicked event.
+   */
+  @Event()
+  clickItem!: EventEmitter<TreeViewItem<string>[]>;
+
+  keyDownListener = (event: KeyboardEvent) => {
+    if (event.defaultPrevented) {
+      return;
+    }
+
+    const isIndexLetter = (str: string) => str.length === 1 && str.match(/\S/);
+
+    const tree = event.composedPath().find(item => (item instanceof HTMLElement) ? item.className === 'dso-tree' : false);
+
+    if (!(event.target instanceof HTMLParagraphElement) || !(tree instanceof HTMLElement)) {
```

Comment:
Ik begrijp waarom je die `!(tree instanceof HTMLElement)` check doet, maar volgens mij kan je op regel 51 ook dit doen:

```
const tree = event.composedPath().find((item): item is HTMLElement => (item instanceof HTMLElement) ? item.className === 'dso-tree' : false);
```

`tree` is dan type `HTMLElement | undefined`.

Reply:
Ik zag niet direct verbetering.